### DISCUSSION TITLE
Custom Server required for ssl

### DISCUSSION THREAD

**Discussion by teremyx** [vercel/next.js#19757] [Help]
Hi. Since I am self-hosting my nextjs-app, I need to use ssl certificates. 
It seems I am required to use a custom server for this. 
Is this not a common scenario for everyone who wants to host the app themselves?
The documentation about custom servers says you should only use if the router does not meet your requirements, but it's actually the ssl requirement that forces me to use a custom server. 
Also I had to configure babel and use ts-node, since it's a typescript project and I want to use es6 modules. 

Am I overlooking something or is this the only way I can configure ssl certificates?

**Comment by arso1er** [comment]
Hi, a reverse proxy (nginx) should be able to get the job done. This medium post should get you started: [https://medium.com/@utkarsh_verma/configure-nginx-as-a-web-server-and-reverse-proxy-for-nodejs-application-on-aws-ubuntu-16-04-server-872922e21d38](https://medium.com/@utkarsh_verma/configure-nginx-as-a-web-server-and-reverse-proxy-for-nodejs-application-on-aws-ubuntu-16-04-server-872922e21d38)

**Reply by teremyx** [reply]
Thanks for the reply. This is however no option, since I would have to use nginx. I just want to use node for the server.
